bin-src/trim-stats-file.ts

```diff
@@ -8,8 +8,12 @@ const isUserCode = ({ name, moduleName = name }: { name?: string; moduleName?: s
   !moduleName.startsWith('(webpack)') &&
   !moduleName.match(/(node_modules|webpack\/runtime)\//);
 
-export const readStatsFile = async (filePath: string): Promise<Stats> =>
-  parseChunked(createReadStream(filePath));
+export const readStatsFile = async (filePath: string): Promise<Stats> => {
+  if (existsSync(filePath)) {
```

Comment:
I added this check because I found that in the `upload.test.ts` test for traced files, it was erroring out because the file path does not return an existing file.

I attempted to mock the file path but because we mock `fs-extra` it overrides all my mock tests and cannot find the file.

The logic check ensures that we don't error out on a stats file with a directory that we cannot process.

However, once the trace or upload reaches the point where it's processing the stats file, we have already checked for a file path and that the file exists so we should be able to reasonably assume that the file exists and this is probably an indicator that the test needs to be refactored.

Reply:
What is strange is that when you include a combination of the `sourceDir` and `statsPath` the upload process builds a path that would most likely not exist in production.
```
sourceDir: '/static/',
fileInfo: { statsPath: '/static/preview-stats.json' },
```
This is the path: `static/static/preview-stats.json`
This is the existing test so I'm hesitant to change it but that file path does not seem like something that would ever exist in a real project.

So is the test incorrect, is the path join incorrect, or is this expected?
@ghengeveld @tmeasday

Reply:
@ethriel3695 reading where `statsPath` comes from, it is relative to the `sourceDir`: https://github.com/chromaui/chromatic-cli/blob/0d1bd21f838decd4709d66db55c525870ab0bef5/bin-src/tasks/upload.ts#L77-L117

So I think that simply the test is wrong here.